### PR TITLE
Bump required node version to >=12.14.0  and build using rollup

### PR DESCRIPTION
**Pull request by BPScott**
## Description

A pair of breaking changes.

- Update required node version to be >=12.14.0, and add an engines.node key to each package.json to enforce that.
- Update sewing-kit-next build config to use the rollup build.

This PR also contains some CI simplifications copied from #1908. Hopefully that can get merged then I'll rebase this PR.

## Type of change

Every package: MAJOR

### CODE REVIEW

File: packages/graphql-tool-utilities/package.json [hunk 1]
```diff
@@ -20,6 +20,9 @@
     "url": "https://github.com/Shopify/quilt/issues"
   },
   "homepage": "https://github.com/Shopify/quilt/blob/main/packages/graphql-tool-utilities/README.md",
+  "engines": {
+    "node": ">=12.14.0"
+  },
   "dependencies": {
     "apollo-codegen-core": "0.36.5",
     "core-js": "^3.0.0",
```
Comment (atesgoral) [inline packages/graphql-tool-utilities/package.json:28]: Remove?

File: gems/quilt_rails/dev.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ up:
   - bundler
   - node:
       yarn: v1.19.1
```
Comment (atesgoral) [inline gems/quilt_rails/dev.yml:6]: Should we bump yarn to 1.22.5 to match the parent dev.yml?

File: gems/quilt_rails/dev.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ up:
...
-      version: v10.18.1
+      version: v12.14.0
```
Comment (atesgoral) [inline gems/quilt_rails/dev.yml:7]: `version` coming after `yarn` is giving me a cognitive dissonance :/ Could we reorder them while we're at it?

**Review by dahukish — APPROVED** [review]
:shipit:

File: packages/graphql-tool-utilities/package.json (hunk 1 above)
Comment (dahukish) [reply packages/graphql-tool-utilities/package.json:28]: This keeps getting added always

Comment (BPScott) [reply packages/graphql-tool-utilities/package.json]: Yep. I removed the usage, but somehow missed the dependency.

**Comment by BPScott** [comment]
Alrighty then. Major release time. Let's do this!

**Pull request merged**